Druid/DRBytecodeGenerator.class.st

```diff
@@ -146,13 +178,32 @@ DRBytecodeGenerator >> visitBlock: aDRBasicBlock [
 { #category : 'ir-to-target' }
 DRBytecodeGenerator >> visitBlockClosure: aDRBlockClosure [
 
-	| blockNode compiledBlock |
+	| blockNode compiledBlock blockCFG blockBuilder copiedValues |
 	blockNode := aDRBlockClosure blockNode.
-	compiledBlock := OCASTTranslator new translateFullBlock: blockNode.
+	blockCFG := aDRBlockClosure generateCFG.
+	
+	blockCFG applyOptimisation: DRFrameInfoCleaner new.
+	blockCFG applyOptimisation: (DRBranchCollapse then: DRPhiSimplication).
+	blockCFG applyOptimisation: (DRSCCP then: DRDeadCodeElimination).
+	DRLocalVariableInstructionScheluder new applyTo: blockCFG.
```

Comment:
This is copy-pasted, we should put it in a good object like `DRMethodCompilerCompiler` and use it